encoding/varint.ts

```diff
@@ -34,22 +38,51 @@ const SHIFTN = 7n;
  * from the returned new `offset`.
  */
 export function decode(buf: Uint8Array, offset = 0): [bigint, number] {
-  for (
-    let i = offset,
-      len = Math.min(buf.length, offset + MaxVarIntLen64),
-      shift = 0,
-      decoded = 0n;
-    i < len;
-    i += 1, shift += SHIFT
-  ) {
-    const byte = buf[i];
-    decoded += BigInt((byte & REST) * Math.pow(2, shift));
-    if (!(byte & MSB) && decoded > MaxUInt64) {
-      throw new RangeError("overflow varint");
+  U64_VIEW[0] = 0n;
+  let intermediate = 0;
+  let position = 0;
+  let i = offset;
+
+  if (buf.length === 0) throw new RangeError("Cannot read empty buffer");
+
```

Comment:
Can you please add a simple test case for this scenario?

Reply:
This should probably be mentioned in a JSDoc imo.